--- httpd-2.4.46/acinclude.m4.detectsystemd	2020-08-07 22:51:40.475205331 +0300
+++ httpd-2.4.46/acinclude.m4	2020-08-07 22:54:08.828106207 +0300
@@ -616,7 +616,7 @@
    elif test -n "$PKGCONFIG" && $PKGCONFIG --exists libsystemd-daemon; then
       SYSTEMD_LIBS=`$PKGCONFIG --libs libsystemd-daemon`
    else
-      AC_CHECK_LIB(systemd-daemon, sd_notify, SYSTEMD_LIBS="-lsystemd-daemon")
+      AC_CHECK_LIB(systemd, sd_notify, SYSTEMD_LIBS="-lsystemd")
    fi
    if test -n "$SYSTEMD_LIBS"; then
       AC_CHECK_HEADERS(systemd/sd-daemon.h)
